### PR TITLE
BACK-1786 Use Account for SyncEvents

### PR DESCRIPTION
**Pull request by estrauser-ledger**
## What is this about?

Account is : 
- [generated] id
- identifier (key)
- coin family
- coin
- group

## Cute picture of lama
lama hairs

### CODE REVIEW

**Comment by mbasire-ledger** [comment]
Why do we need to add account into SyncEvent ?

**Comment by estrauser-ledger** [comment]
So we can have this information in the Worker

File: account-manager/src/main/resources/db/migration/V9__sync_event_with_account_info.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE account_sync_event
+    ADD COLUMN "key" VARCHAR NOT NULL DEFAULT '',
+    ADD COLUMN coin_family coin_family NOT NULL DEFAULT 'bitcoin',
+    ADD COLUMN coin coin NOT NULL DEFAULT 'btc',
```
Comment (gagbo) [inline account-manager/src/main/resources/db/migration/V9__sync_event_with_account_info.sql:4]: Why is there duplicated column names here ?

Comment (estrauser-ledger) [reply account-manager/src/main/resources/db/migration/V9__sync_event_with_account_info.sql:4]: coin_family (protocol) => bitcoin, ethereum, ...
coin => btc, ltc, doge, ...

File: common/src/main/scala/co/ledger/lama/common/models/messages/WorkerMessage.scala [hunk 2]
```diff
@@ -5,11 +5,11 @@ import java.util.UUID
 import io.circe.generic.extras.semiauto._
 import io.circe.{Decoder, Encoder}
 
-import co.ledger.lama.common.models.{AccountIdentifier, WorkableEvent}
+import co.ledger.lama.common.models.{Account, WorkableEvent}
 import co.ledger.lama.common.models.implicits._
 
-// TODO: add account group to business ID ?
-case class WorkerMessage[T](account: AccountIdentifier, event: WorkableEvent[T])
+// TODO: Is this still needed ?
```
Comment (nukky94) [inline common/src/main/scala/co/ledger/lama/common/models/messages/WorkerMessage.scala:11]: Nope :)

**Review by nukky94 — APPROVED** [review]
LGTM but if you want you can remove WorkerMessage and use directly WorkableEvent since we have now the account into it

File: protobuf/src/main/protobuf/account-manager.proto [hunk 3]
```diff
@@ -104,13 +100,20 @@ message GetSyncEventsResult {
...
-    bytes account_id = 1;
+    Account account = 1;
     bytes sync_id = 2;
     string status = 3;
-    bytes cursor = 4;
-    bytes error = 5;
-    google.protobuf.Timestamp time = 6;
+    bytes cursor = 5;
+    bytes error = 6;
+    google.protobuf.Timestamp time = 7;
```
Comment (gagbo) [inline protobuf/src/main/protobuf/account-manager.proto:116]: 1 2 3 5 6 7

File: common/src/main/scala/co/ledger/lama/common/models/messages/WorkerMessage.scala (hunk 2 above)
Comment (estrauser-ledger) [reply common/src/main/scala/co/ledger/lama/common/models/messages/WorkerMessage.scala:11]: This is not a small change, I'll create a ticket

**Pull request merged**